### ISSUE TITLE
Compile TensorFlow Lite for iOS Simulator on Apple Silicon

### ISSUE THREAD

**Issue by bvirlet** [tensorflow/tensorflow#47400]
TensorFlowLite isn't available for iOS Simulator running on Apple Silicon.

It seems that this depends on [Bazel supporting this](https://github.com/bazelbuild/rules_apple/issues/980), as well as on TensorFlowLite being built as a .xcframework (fat frameworks have an arm64 slice, but cannot have an arm64-mac slice).

**Comment by abattery** [comment]
@yyoon could you take a look?

**Comment by yyoon** [comment]
@bvirlet Yes, you are correct that we're blocked by the missing bazel support. Once bazel supports it, we will be migrating to .xcframework to incorporate the different target architectures, including the simulators on the new Apple silicon.

**Comment by bvirlet** [comment]
Thanks. For reference, I'm linking this issue here that's related/the same: https://github.com/tensorflow/tensorflow/issues/44609

**Comment by greenzeal** [comment]
+1

**Comment by bvirlet** [comment]
@yyoon do you think [this PR](https://github.com/bazelbuild/bazel/pull/12900) is what we want from Bazel?

**Comment by yyoon** [comment]
@bvirlet I've looked at that PR before, but that's for building bazel itself to work natively on M1, which is different from building an ios_static_framework for arm64 simulator target. I think this would involve more work from apple build rules side.

Even better, if bazel + apple rules start to support building an XCFramework with multiple target architectures, we could switch to using that for releasing our iOS binaries.

**Comment by bvirlet** [comment]
Thanks for clarifying that!

**Comment by guillaume-tgl** [comment]
Looks like a new version of Bazel's `rules_apple` has been released with support for `ios_sim_arm64` architecture: https://github.com/bazelbuild/rules_apple/releases/tag/0.32.0

Is it enough to be able to build TFLite for iOS Simulator on Apple Silicon?

**Comment by yyoon** [comment]
That's good to hear. Roughly speaking, this could be done in the following steps.

1. Update bazel to 5.0.0 or higher.
2. Update `rules_apple` version to 0.32.0 or higher. This version dependency is defined in: https://github.com/tensorflow/tensorflow/blob/592f3797f814eb280eb39bda8aea193e5d5c60bc/tensorflow/workspace2.bzl#L801-L805
3. Add a new `.bazelrc` config for `ios_sim_arm64`: https://github.com/tensorflow/tensorflow/blob/592f3797f814eb280eb39bda8aea193e5d5c60bc/.bazelrc#L193-L206
4. Build the iOS static famework following [this guide](https://www.tensorflow.org/lite/guide/build_ios#build_tensorflowlitec_dynamic_framework_recommended) but with `--config=ios_sim_arm64` option instead.

I didn't have time to try this out myself unfortunately and there might be other roadblocks ahead, but it'd be worth trying if you're in need.

**Comment by edritech93** [comment]
same issue for me

**Comment by snyuryev** [comment]
@yyoon 
Tried you suggestion and got this error:
```
ERROR: Config expansion has a cycle: config value ios_sim_arm64 expands to itself, see inheritance chain [ios_sim_arm64]
```

Any updates on this issue? What is the way to compile for arm64 simulator?

**Comment by yyoon** [comment]
@yishuangP Could you take a look when you get a chance? Thanks.

**Comment by JackTudor96** [comment]
Hi guys, we're using a 3rd party lib that uses TF Lite and we're running into this issue. Do we have an idea when this is likely to be picked up?

**Comment by mself** [comment]
This is close to working since Bazel 5 is now supported and 2.9 already includes `rules_apple` 0.33.  But I am still getting an error due to the bundled copy of `cpuinfo.BUILD`, which doesn't have `ios_sim_arm64` in it?

**Comment by mself** [comment]
Thanks, @ccarpita.  I verified that your commit works for my project.

**Comment by mself** [comment]
This issue can be closed now that #55777 has been merged.

**Comment by yishuangP** [comment]
FYI TensorFlow Lite also starts distributing XCFrameworks that has an arm64 slice through Cocoapods in nightly releases https://github.com/CocoaPods/Specs/tree/master/Specs/1/6/0/TensorFlowLiteC/0.0.1-nightly.20220513.

**Comment by snyuryev** [comment]
@yishuangP thank you!

**Comment by PhilipDukhov** [comment]
I was able to built it for simulator using command from #55777, but it produces simulator only framework, which cannot be used to run on a device. 
`ios_fat` [doesn't have](https://github.com/tensorflow/tensorflow/blob/d8ce9f9c301d021a69953134185ab728c1c248d3/.bazelrc#L205) `ios_sim_arm64` in the list, I've tried adding it manually but the build fails.

How fat framework can be built to support both simulator and device?
